src/lib/registration/rollback_script.rb

```diff
@@ -0,0 +1,72 @@
+# ------------------------------------------------------------------------------
+# Copyright (c) 2018 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# ------------------------------------------------------------------------------
+
+require "yast"
+require "fileutils"
+
+module Registration
+  # This class handles creating a registration rollback script  which is called
+  # when the upgrade is aborted or YaST crashes. The script body is stored
+  # in the data/registration/registration_rollback.sh file.
+  class RollbackScript
+    include Yast::Logger
+
+    BACKUP_DIR = "var/adm/backup/system-upgrade".freeze
+
+    # use number 0200, the original repositories need to be restored first,
+    # they are stored in script with number 0100
+    DEFAULT_SCRIPT_NAME = "restore-0200-registration.sh".freeze
+
+    SUSE_CONNECT = "/usr/sbin/SUSEConnect".freeze
+
+    attr_reader :name, :root
+
+    # @param name [String] script name
+    # @param root [String] target root
+    def initialize(name: DEFAULT_SCRIPT_NAME, root: "/mnt")
```

Comment:
what is usecase for diferent script name? I do not see it used, so for simplicify of API I will just remove it.